agrona/src/main/java/org/agrona/concurrent/status/AtomicCounter.java

```diff
@@ -249,6 +268,27 @@ public long decrementOrdered()
         return currentValue;
     }
 
+    /**
+     * Decrements the counter.
+     * <p>
+     * This method is not atomic and this can lead to lost-updates due to race conditions. This load and store
+     * have plain memory semantics.
+     * <p>
+     * The typical use-case for this method is when writer and reader are the same thread.
+     *
+     * @return the previous value of the counter
+     * @since 2.1.0
+     */
+    public long decrementPlain()
+    {
+        final byte[] array = byteArray;
+        final long offset = addressOffset;
+        final long currentValue = UnsafeApi.getLong(array, offset);
+        UnsafeApi.putLong(array, offset, currentValue - 1);
+
+        return currentValue;
+    }
+
```

Comment:
Should just call a helper method to be used for both `increment` and `decrement` operations. That method needs a CAS loop (see above).

Reply:
I'm not sure about this part. Helper methods lead to another method call and there is a limit on the number of method call the JIT will inline. So if you insist I'll will add the helper method, but from a performance perspective, the duplication can lead to faster code.

Reply:
No need to.